compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/viewinterop/UIKitInteropElementHolder.uikit.kt

```diff
@@ -114,17 +152,40 @@ internal abstract class UIKitInteropElementHolder<T : InteropView>(
 
             // update the user component frame only if it changes
             if (userComponentRect != currentUserComponentRect) {
-                currentUserComponentRect = userComponentRect
-
-                val userComponentFrame =
+                // Schedule frame update
+                val newUserComponentCGRect =
                     userComponentRect
                         .toRect()
                         .toDpRect(density)
                         .asCGRect()
 
                 container.scheduleUpdate {
-                    setUserComponentFrame(userComponentFrame)
+                    userComponentCGRect = newUserComponentCGRect
+                }
+
+                // Check if component size changed
+                val newUserComponentSize =
+                    if (userComponentRect.size != currentUserComponentRect?.size) {
+                        userComponentRect.size
+                    } else {
+                        null
+                    }
+
+                // Schedule invoking callbacks with updated data
+                callbacks?.run {
+                    newUserComponentSize?.let {
```

Comment:
Nitpick: just regular `if`? double lambda wrapping seems weird

Reply:
Yeah, it's a side effect of my experimentation with reassignable callbacks/properties and callbacks being `var`. Changed it.